src/services/ai_service.py

```diff
@@ -1,66 +1,89 @@
-import json
-from typing import List, Dict, Any, Optional
-import google.generativeai as Client
+from typing import List, Dict
 from unidiff import Hunk, PatchedFile
 from ..core.config import Config
-from ..core.models import PRDetails, FileInfo
-
+from ..core.models import PRDetails
+from .llms.gemini import GeminiService
+from .llms.openai import OpenAIService
 
 class AIService:
-  def __init__(self):
-    self.model = Client.GenerativeModel(Config.GEMINI_MODEL)
-
-  def create_prompt(
-    self, file: PatchedFile, hunk: Hunk, pr_details: PRDetails
-  ) -> str:
-    return f"""Your task is reviewing pull requests. Instructions:
```

Comment:
Consider adding more robust error handling and logging for `_initialize_service`.  Currently, it prints error messages to the console, but in a production environment, more sophisticated logging (e.g., using a logging library) would be beneficial for debugging and monitoring.